src/AISmart.GAgent.Twitter/Provider/TwitterProvider.cs

```diff
@@ -3,126 +3,201 @@
+    private readonly IOptionsMonitor<TwitterOptions> _twitterOptions;
+    private readonly HttpClient _httpClient;
+    private readonly AESCipher _aesCipher;
+    
+    public TwitterProvider(ILogger<ITwitterProvider> logger, IOptionsMonitor<TwitterOptions> twitterOptions)
     {
         _logger = logger;
-        _telegramOptions = telegramOptions;
+        _twitterOptions = twitterOptions;
+        _httpClient = new HttpClient();
+        string password = twitterOptions.CurrentValue.EncryptionPassword;
+        _aesCipher = new AESCipher(password);
     }
-
-    public async Task<List<Tweet>> GetLatestTwittersAsync(string sendUser, string userId, string sinceTweetId)
+    
+    public async Task<List<Tweet>> GetMentionsAsync()
     {
-        var bearerToken = GetAccount(sendUser);
-        string url = $"https://api.twitter.com/2/tweets"; 
-        int maxResults = 3; 
-        _logger.LogInformation($"twitterToken = {bearerToken}, userId = {userId},sinceTweetId = {sinceTweetId}");
+        var bearerToken = _twitterOptions.CurrentValue.BearerToken;
+        string username = "elonmusk";
+        string query = $"@{username}";
+        string encodedQuery = Uri.EscapeDataString(query);
+        string url = $"https://api.twitter.com/2/tweets/search/recent?query={encodedQuery}&tweet.fields=author_id,conversation_id&max_results=100";
 
         using (var client = new HttpClient())
         {
             client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", bearerToken);
-            var requestUrl = $"{url}?max_results={maxResults}&since_id={{sinceTweetId}}";
-            var response = await client.GetAsync(requestUrl);
-          
-            if (response.Headers.Contains("x-rate-limit-limit"))
-            {
-                var rateLimit = response.Headers.GetValues("x-rate-limit-limit");
-                var rateLimitRemaining = response.Headers.GetValues("x-rate-limit-remaining");
-                var rateLimitReset = response.Headers.GetValues("x-rate-limit-reset");
-
-                _logger.LogInformation($"rateLimit = {rateLimit.First()}, rateLimitRemaining = {rateLimitRemaining.First()},rateLimitReset = {rateLimitReset.First()}");
-                if (rateLimitRemaining.First() == "0")
-                {
-                    var resetTime = DateTimeOffset.FromUnixTimeSeconds(long.Parse(rateLimitReset.First()));
-                    var waitTime = resetTime - DateTimeOffset.UtcNow;
-                    _logger.LogInformation($"waitTime {waitTime.TotalSeconds} ");
-                }
-            }
+            
+            var response = await client.GetAsync(url);
 
             if (response.IsSuccessStatusCode)
             {
                 string responseBody = await response.Content.ReadAsStringAsync();
                 _logger.LogInformation("responseBody: " + responseBody);
                 var responseData = JsonConvert.DeserializeObject<TwitterResponseDto>(responseBody);
```

Comment:
please always add exceptions. Check through your code to make sure that exceptions are caught. If any exception isn't caught, it might crash the silo. Be detailed in your code in catching exceptions